tilings/strategies/sliding.py

```diff
@@ -0,0 +1,286 @@
+        return GriddedPerm(gp.patt.rotate(-1), ((0, x) for x, _ in gp.pos))
+
+    return _tmp_func
+
+
+def _gp_rotate90_and_reverse(r: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(
+            gp.patt.rotate(1).reverse(), ((r - y - 1, 0) for _, y in gp.pos)
+        )
+
+    return _tmp_func
+
+
+def _gp_rotate90_and_reverse_inverse(r: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(
+            gp.patt.reverse().rotate(-1), ((0, r - x - 1) for x, _ in gp.pos)
+        )
+
+    return _tmp_func
+
+
+class _AdditionalMaps:
+    def __init__(
+        self,
+        identifier: int = 0,
+        param: int = 0,
+        t_inv: Callable[[Tiling], Tiling] = _tiling_identity_map,
+        g_map: Callable[[GriddedPerm], GriddedPerm] = _gp_identity_map,
+        g_inv: Callable[[GriddedPerm], GriddedPerm] = _gp_identity_map,
+    ) -> None:
+        self.identifier = identifier
+        self.param = param
+        self.t_inv = t_inv
+        self.g_map = g_map
+        self.g_inv = g_inv
+
+    @classmethod
+    def reverse(cls, c: int) -> "_AdditionalMaps":
+        return _AdditionalMaps(
+            1,
+            c,
+            Tiling.reverse,
+            _gp_reverse(c),
+            _gp_reverse(c),
+        )
+
+    @classmethod
+    def rotate90(cls, r: int) -> "_AdditionalMaps":
+        return _AdditionalMaps(
+            2,
+            r,
+            Tiling.rotate270,
+            _gp_rotate90(r),
+            _gp_rotate90_inverse(r),
+        )
+
+    @classmethod
+    def rotate90_and_reverse(cls, r: int) -> "_AdditionalMaps":
+        return _AdditionalMaps(
+            3,
+            r,
+            Tiling.antidiagonal,
+            _gp_rotate90_and_reverse(r),
+            _gp_rotate90_and_reverse_inverse(r),
+        )
+
+    def to_jsonable(self) -> dict:
+        """Return a dictionary form of the strategy."""
+        return {"identifier": self.identifier, "param": self.param}
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "_AdditionalMaps":
+        """Create instance from dictionary."""
+        _id = d["identifier"]
+        if _id == 0:
+            return cls()
+        if _id == 1:
+            return cls.reverse(d["param"])
+        if _id == 2:
+            return cls.rotate90(d["param"])
+        return cls.rotate90_and_reverse(d["param"])
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, type(self)):
+            return False
+        return self.identifier == other.identifier
+
+
+class SlidingStrategy(DisjointUnionStrategy[Tiling, GriddedPerm]):
+    """
+    A class for a specific slidding strategy. The init gets the index of both
+    column you slidding.
+    """
+
+    def __init__(
+        self,
+        av_12_column: int,
+        av_123_column: int,
+        sliding: Sliding,
```

Comment:
Isn't that a bit too much that the strategy needs to store the whole sliding algorithm? If I compare with the fusion algorithm, only the column or row fused is stored and the algorithm can be recomputed if needed.